rholang/src/test/scala/coop/rchain/rholang/interpreter/ShortCircuitBooleanSpec.scala

```diff
@@ -0,0 +1,172 @@
+  implicit val noopSpan: Span[Task]       = NoopSpan[Task]()
+  private val maxDuration                 = 5.seconds
+
+  val outcomeCh      = "ret"
+  val reduceErrorMsg = "Error: index out of bound: -1"
+
+  private def execute(source: String): Task[Either[InterpreterError, Boolean]] =
+    mkRuntime[Task]("rholang-short-circuit-boolean")
+      .use { runtime =>
+        for {
+          evalResult <- runtime.evaluate(source)
+          result <- if (evalResult.errors.isEmpty)
+                     for {
+                       data       <- runtime.getData(GString(outcomeCh)).map(_.head)
+                       boolResult = data.a.pars.head.exprs.head.getGBool
+                     } yield Right(boolResult)
+                   else Task.pure(Left(evalResult.errors.head))
+        } yield result
+      }
+
+  " `par1 && par2` rholang boolean evaluation" should {
+    "execute only the first par1 if par1==false" in {
+      val term =
+        s"""
+            # @"${outcomeCh}"!(false && [1,2].nth(-1))
+            # """.stripMargin('#')
+      execute(term).runSyncUnsafe(maxDuration) should equal(Right(false))
+
+      val term2 =
+        s"""
+            # @"${outcomeCh}"!(1 < 0 && [1,2].nth(-1))
+            # """.stripMargin('#')
+      execute(term2).runSyncUnsafe(maxDuration) should equal(Right(false))
+    }
+    "execute both par1 and par2 if par1 == true" in {
+      val term =
+        s"""
+           # @"${outcomeCh}"!(true && [1,2].nth(-1))
+           # """.stripMargin('#')
+
+      execute(term).runSyncUnsafe(maxDuration) should equal(Left(ReduceError(reduceErrorMsg)))
+
+      val term2 =
+        s"""
+           # @"${outcomeCh}"!(1>0 && [1,2].nth(-1))
+           # """.stripMargin('#')
+
+      execute(term2).runSyncUnsafe(maxDuration) should equal(Left(ReduceError(reduceErrorMsg)))
+    }
+  }
+
+  " `par1` || par2` rholang boolean evaluation" should {
+    "evaluate only the first par1 if par1==true" in {
+      val term =
+        s"""
+           # @"${outcomeCh}"!(true || [1,2].nth(-1))
+           # """.stripMargin('#')
+      execute(term).runSyncUnsafe(maxDuration) should equal(Right(true))
+
+      val term2 =
+        s"""
+           # @"${outcomeCh}"!(1 > 0 || [1,2].nth(-1))
+           # """.stripMargin('#')
+      execute(term2).runSyncUnsafe(maxDuration) should equal(Right(true))
+    }
+    "evaluate both par1 and par2 if par1 == false" in {
+      val term =
+        s"""
+           # @"${outcomeCh}"!(false || [1,2].nth(-1))
+           # """.stripMargin('#')
+
+      execute(term).runSyncUnsafe(maxDuration) should equal(Left(ReduceError(reduceErrorMsg)))
+
+      val term2 =
+        s"""
+           # @"${outcomeCh}"!(1<0 || [1,2].nth(-1))
+           # """.stripMargin('#')
+
+      execute(term2).runSyncUnsafe(maxDuration) should equal(Left(ReduceError(reduceErrorMsg)))
+    }
+  }
+
+  "&&" should {
+    "work good with `and`" in {
```

Comment:
_Work good_ is very general description so there is no indication what is tested.